tripy/tripy/frontend/utils.py

```diff
@@ -84,11 +85,23 @@ def _add_column_info(arg, arg_index, is_kwarg, num_positional, func_name, skip_n
 
     # Find the first caller of this function that is NOT the function registry.
     # Also save the last dispatch target we see.
+
+    # Start from the registry. It will always be present except for tests,
+    # since the decorator is intended only for overrides of magic functions.
+    # This check supports cases like slice_helper, where the decorated function
+    # is used *inside* the override and hence the wrapped call would come *before*
+    # the call from the registry.
+    REGISTRY_STACK_DEPTH = WRAPPER_STACK_DEPTH
```

Comment:
Do we still need to find `WRAPPER_STACK_DEPTH` first?

Reply:
It would only be needed in our test cases, unless we want to add those to the function registry.

Reply:
I would slightly prefer adding hacks into the tests rather than into Tripy proper.

Reply:
I'll change the tests, in that case.

Reply:
Hm, it seems my assumption that it's for overrides is incorrect. `convert_to_tensors` is also used on `full`, `full_like`, `iota`, `quantize`, `dequantize`, `expand`, `reshape`, and `resize`, though these appear in a function registry via `export.public_api`, so the check still succeeds.

Reply:
The code actually proves to be necessary in the following case: If `tensor_from_shape_like` is used by the decorator, that may call `concatenate`, which is in the function registry due to the use of `export.public_api`. Without the earlier loop, the inner call to `concatenate` is selected first and the stack entry is chosen. This arises in various test cases that use `tensor_initializers`, e.g., `test_unsqueeze`. I will see if that case can be excluded cleanly.

Reply:
I think skipping stack entries is fundamentally broken. Unless the function signature is the same, it's difficult to reliably determine column information. Could we just update the slice implementation to add column information without the help of this decorator and then disallow skipping stack entries here?

Reply:
Okay, I'll keep the old behavior of checking for the call to the decorator.

Reply:
Updated, probably fewer hacks needed overall